internal/resources/azure/sql_managed_instance.go

```diff
@@ -0,0 +1,196 @@
+package azure
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+const (
+	sqlMIServiceName   = "SQL Managed Instance"
+	sqlMIProductFamily = "Databases"
+)
+
+// SQLManagedInstance struct represents an azure Sql Managed Instance.
+//
+// The azurerm_sql_managed_instance resource is deprecated in version 3.0 of the AzureRM provider and will be removed in version 4.0.
+// Please use the azurerm_mssql_managed_instance resource instead when available in infracost
+//
+// Only support for Gen5 is available on that resource
+//
+// More resource information here: https://azure.microsoft.com/en-gb/products/azure-sql/managed-instance/
+// Pricing information here: https://azure.microsoft.com/en-gb/pricing/details/azure-sql-managed-instance/single/
+type SQLManagedInstance struct {
+	Address            string
+	Region             string
+	SKU                string
+	LicenceType        string
+	Cores              *int64
+	StorageSizeInGb    *int64
```

Comment:
again `StorageSizeInGb` should not be a pointer. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/sql_managed_instance#storage_size_in_gb is marked as required - so it should always be set.